docs/api.md

```diff
@@ -0,0 +1,447 @@
+```
+
+The `evt.initiator` value is a string indicating what triggered the suspended state.
+Possible values: `network`, `manual`.
+
+[Back to top](#api-documentation)
+
+### Event: `resumed`
+
+Handle resumed state. This event is triggered when the session was properly resumed.
+It is useful in scenarios where you for example can close blocking modal dialogs
+and allow the user to interact with your application again.
+
+Example:
+
+```js
+session.on('resumed', () => console.log('The session was resumed'));
+```
+
+[Back to top](#api-documentation)
+
+### Event: `handle:changed`
+
+This event is triggered when QIX Engine notifies us about a QIX handle being changed.
+This means that we may fetch new information from the object (layout, properties, etc.).
+
+Mainly useful in debugging purposes, use the generated API's [`changed`](#event-changed)
+event instead.
+
+Example:
+
+```js
+session.on('handle:changed', (handle) => console.log(`Handle ${handle} was changed`));
+```
+
+[Back to top](#api-documentation)
+
+### Event: `handle:closed`
+
+This event is triggered when QIX Engine notifies us about a QIX handle being closed.
+This means that we may clean up any caches on our side.
+
+Mainly useful in debugging purposes, use the generated API's [`closed`](#event-closed-1)
+event instead.
+
+Example:
+
+```js
+session.on('handle:closed', (handle) => console.log(`Handle ${handle} was closed`));
+```
+
+[Back to top](#api-documentation)
+
+### Event: `notification:<name>`
+
+Handle a specific JSON-RPC notification event. These events depend on the product
+you use QIX Engine from.
+
+TODO: Insert links to Qlik help.
+
+Example:
+
+```js
+session.on('notification:OnConnected', (data) => console.log(data));
+``` 
+
+[Back to top](#api-documentation)
+
+### Event: `notification:*`
+
+Handle all JSON-RPC notification events.
+
+Example:
+
+```js
+session.on('notification:*', (eventName, data) => console.log(eventName, data));
+``` 
+
+[Back to top](#api-documentation)
+
+### Event: `traffic:sent`
+
+Handle outgoing websocket messages.
+
+Generally used in debugging purposes.
+
+Example:
+
+```js
+session.on('traffic:sent', (req) => console.log(req));
+``` 
+
+[Back to top](#api-documentation)
+
+### Event: `traffic:received`
+
+Handle incoming websocket messages.
+
+Generally used in debugging purposes.
+
+Example:
+
+```js
+session.on('traffic:received', (res) => console.log(res));
+``` 
+
+[Back to top](#api-documentation)
+
+### Event: `traffic:*`
+
+Handle all websocket messages.
+
+Generally used in debugging purposes.
+
+Example:
+
+```js
+session.on('traffic:*', (direction, msg) => console.log(direction, msg));
+```
+
+[Back to top](#api-documentation)
+
+## Generated API
+
+The API for generated APIs depends on the QIX Engine schema you pass into your
+[Configuration](#configuration), and on what QIX struct the API has.
+
+Read more: [Generic object model](./concepts.md#generic-object-model)
+
+Example:
+
```

Comment:
Good question! I imagined us writing a highlights file for the new major version highlighting the features etc., but I think it would be good to mention it somewhere here too! I'll look into it.